master/getting-started/kubernetes/installation/migration-from-flannel.md

```diff
@@ -0,0 +1,155 @@
+
+   Once applied, you will see the nodes begin to update one at a time.
+
+1. To monitor the migration, run the following command.
+
+   ```
+   kubectl get jobs -n kube-system flannel-migration
+   ```
+
+   Note that the migration controller may be rescheduled one or more times during the migration as the node hosting
+   the controller is upgraded. The installation will be complete when the ouput of the above command shows `1/1` completions. For example:
+
+   ```
+   NAME                COMPLETIONS   DURATION   AGE
+   flannel-migration   1/1           2m59s      5m9s
+   ```
+
+1. Once complete, remove the migration controller with the following command.
+
+   ```
+   kubectl delete -f {{site.url}}/{{page.version}}/manifests/flannel-migration/migration-job.yaml
+   ```
+# Configuration options
+
+The migration controller will auto-detect your flannel configuration, and in most cases does not require any
+additional configuration. If you do require special configuration, the controller exposes the following options
+which can be set as environment variables within the pod.
+
+| Configuration options            | Description                                                          | Default                                    |
+|----------------------------------|----------------------------------------------------------------------|--------------------------------------------|
+| FLANNEL_NETWORK                  | IPv4 network CIDR used by flannel for the cluster.                   | <Automatically detected>                   |
+| FLANNEL_DAEMONSET_NAME           | Name of the flannel daemon set in the kube-system namespace.         | kube-flannel-ds-amd64                      |
+| FLANNEL_MTU                      | MTU for the flannel VXLAN device.                                    | <Automatically detected>                   |
+| FLANNEL_IP_MASQ                  | Whether masquerading is enabled for outbound traffic.                | true                                       |
+| FLANNEL_SUBNET_LEN               | Per-node subnet length used by flannel.                              | 24                                         |
+| FLANNEL_ANNOTATION_PREFIX        | Value provided via the kube-annotation-prefix option to flannel.     |  flannel.alpha.coreos.com                  |
+| FLANNEL_VNI                      | The VNI used for the flannel network.                                |  flannel.alpha.coreos.com                  |
+| FLANNEL_PORT                     | UDP port used for VXLAN.                                             |  8472                                      |
+| CALICO_DAEMONSET_NAME            | Name of the calico daemon set in the kube-system namespace.          |  calico-node                               |
+| CNI_CONFIG_DIR                   | Full path on the host in which to search for CNI config files.       |  calico-node                               |
+
+# Troubleshooting
+
+## Checking migration status
```

Comment:
## Checking migration status

 The migration controller should run to completion. You can run the following command to view the controller's current status.

 ```
kubectl get pods -n kube-system -l k8s-app=flannel-migration-controller
```

 ## Viewing migration logs

 The migration controller logs can be viewed with the following command:

 ```
kubectl logs -n kube-system -l k8s-app=flannel-migration-controller
```

 The logs will indicate if any actions need to be taken by the user.

 # Rollback

 Migration from Calico to flannel is not supported. If a problem is encountered during the migration, you can use the following steps
to rollback the migration.

 1. Remove the migration controller and Calico.

    ```
   kubectl delete -f {{site.url}}/{{page.version}}/manifests/flannel-migration/migration-job.yaml
   kubectl delete -f {{site.url}}/{{page.version}}/manifests/flannel-migration/calico.yaml
   ```

 1. Determine the nodes which have been migrated to Calico.

    ```
   kubectl get nodes -l projectcalico.org/node-network-during-migration=calico
   ```

 Then, for each node found above, run the following commands to remove Calico.

 1. Cordon and drain the node.

    ```
   kubectl drain node <node name>
   ```

 1. Log into the machine and remove the CNI configuration.

    ```
   rm /etc/cni/net.d/10-calico.conflist
   ```

 1. Reboot the node.

 1. Enable flannel on the node.

    ```
   kubectl label node <node name> projectcalico.org/node-network-during-migration=flannel
   ```

 1. Uncordon the node.

    ```
   kubectl uncordon node <node name>